lms/djangoapps/teams/tests/test_api.py

```diff
@@ -81,34 +101,48 @@ def test_public_discussion_visible_by_user(self):
         self.assertTrue(teams_api.discussion_visible_by_user(self.team2.discussion_topic_id, self.user2))
         self.assertTrue(teams_api.discussion_visible_by_user('DO_NOT_EXISTS', self.user3))
 
-    def test_get_team(self):
-        # Course 1
-        user1_team = teams_api.get_team_for_user_and_course(self.user1, str(COURSE_KEY1))
-        user2_team = teams_api.get_team_for_user_and_course(self.user2, str(COURSE_KEY1))
-        user3_team = teams_api.get_team_for_user_and_course(self.user3, str(COURSE_KEY1))
-
-        self.assertEqual(user1_team, self.team1)
-        self.assertEqual(user2_team, self.team1)
-        self.assertEqual(user3_team, None)
-
-        # Course 2
-        user1_team = teams_api.get_team_for_user_and_course(self.user1, str(COURSE_KEY2))
-        user2_team = teams_api.get_team_for_user_and_course(self.user2, str(COURSE_KEY2))
-        user3_team = teams_api.get_team_for_user_and_course(self.user3, str(COURSE_KEY2))
+    @ddt.unpack
+    @ddt.data(
+        (COURSE_KEY1, TOPIC1, ['team1', 'team1', None, 'team1a']),
+        (COURSE_KEY1, TOPIC2, [None, None, None, None]),
+        (COURSE_KEY2, TOPIC1, [None, None, None, None]),
+        (COURSE_KEY2, TOPIC2, [None, None, 'team2', 'team2a']),
+    )
+    def test_get_team_for_user_course_topic(self, course_key, topic_id, expected_team_ids):
+        user1_team = teams_api.get_team_for_user_course_topic(self.user1, str(course_key), topic_id)
+        user2_team = teams_api.get_team_for_user_course_topic(self.user2, str(course_key), topic_id)
+        user3_team = teams_api.get_team_for_user_course_topic(self.user3, str(course_key), topic_id)
+        user4_team = teams_api.get_team_for_user_course_topic(self.user4, str(course_key), topic_id)
+
+        self.assertEqual(user1_team.team_id if user1_team else None, expected_team_ids[0])
+        self.assertEqual(user2_team.team_id if user2_team else None, expected_team_ids[1])
+        self.assertEqual(user3_team.team_id if user3_team else None, expected_team_ids[2])
+        self.assertEqual(user4_team.team_id if user4_team else None, expected_team_ids[3])
+
+    @mock.patch('lms.djangoapps.teams.CourseTeam.objects')
+    def test_get_team_multiple_teams(self, mocked_manager):
+        """
+        This is a test for a use case that should never happen.
+        Currently users cannot be in multiple teams in a course, but even after we allow multiple
+        teams in a course then they should still be limited to one team per topic
+        """
```

Comment:
I understand wanting to cover all the bases, but not sure about testing something that should never happen? Perhaps we shouldn't say "never" and instead call it a low probability event?